src/main/java/com/google/api/gax/grpc/RpcFuture.java

```diff
@@ -42,4 +42,11 @@
   void addListener(Runnable listener, Executor executor);
 
   void addCallback(RpcFutureCallback<? super V> callback);
+
+  <X extends Throwable> RpcFuture catching(
+      Class<X> exceptionType, Function<? super X, ? extends V> callback);
+
+  interface Function<F, T> {
```

Comment:
@garrettjonesgoogle Do you want this moved out into its own class. I'm not against it but I personally find it confusing to have a lot of top-level classes. Might just be me though